comps/dataprep/src/integrations/mariadb.py

```diff
@@ -0,0 +1,386 @@
+            doc_path = self.upload_folder + link + ".txt"
+            if logflag:
+                logger.info(f"[ ingest link ] save_path: {save_path}")
+            await save_content_to_local_disk(save_path, content)
+            metadata = [dict({"doc_name": str(doc_path)})]
+
+            chunks = text_splitter.split_text(content)
+
+            batch_size = 32
+            num_chunks = len(chunks)
+            for i in range(0, num_chunks, batch_size):
+                batch_chunks = chunks[i : i + batch_size]
+                batch_texts = batch_chunks
+
+                _= MariaDBStore.from_texts(
+                    texts=batch_texts,
+                    embedding=self.embedder,
+                    embedding_length=EMBEDDING_LENGHT,
+                    metadatas=metadata,
+                    collection_name=INDEX_NAME,
+                    datasource=MARIADB_CONNECTION_STRING,
+                )
+                if logflag:
+                    logger.info(f"Processed batch {i//batch_size + 1}/{(num_chunks-1)//batch_size + 1}")
+
+        return True
+
+    async def ingest_files(
+        self,
+        files: Optional[Union[UploadFile, List[UploadFile]]] = File(None),
+        link_list: Optional[str] = Form(None),
+        chunk_size: int = Form(1500),
+        chunk_overlap: int = Form(100),
+        process_table: bool = Form(False),
+        table_strategy: str = Form("fast"),
+        ingest_from_graphDB: bool = Form(False),
+    ):
+        """Ingest files/links content into database.
+
+        Save in the format of vector[768].
+        Returns '{"status": 200, "message": "Data preparation succeeded"}' if successful.
+        Args:
+            files (Union[UploadFile, List[UploadFile]], optional): A file or a list of files to be ingested. Defaults to File(None).
+            link_list (str, optional): A list of links to be ingested. Defaults to Form(None).
+            chunk_size (int, optional): The size of the chunks to be split. Defaults to Form(1500).
+            chunk_overlap (int, optional): The overlap between chunks. Defaults to Form(100).
+            process_table (bool, optional): Whether to process tables in PDFs. Defaults to Form(False).
+            table_strategy (str, optional): The strategy to process tables in PDFs. Defaults to Form("fast").
+        """
+        if logflag:
+            logger.info(f"files:{files}")
+            logger.info(f"link_list:{link_list}")
+        if files and link_list:
+            raise HTTPException(status_code=400, detail="Provide either a file or a string list, not both.")
+
+        if files:
+            if not isinstance(files, list):
+                files = [files]
+
+            if not os.path.exists(self.upload_folder):
+                Path(self.upload_folder).mkdir(parents=True, exist_ok=True)
+            for file in files:
+                save_path = self.upload_folder + file.filename
+                await self.save_file_to_local_disk(save_path, file)
```

Comment:
If we're going to use Path above, let's clean this up:

```python
upload_folder_path = Path(self.upload_folder)
for file in files:
    save_path = upload_folder_path / file.filename
    await self._save_file_to_local_disk(save_path, file)
```

And change save_file_to_local_disk to:

```python
    async def _save_file_to_local_disk(self, save_path: Path, file: File):
        with save_path.open("wb") as fout:
            try:
                content = await file.read()
                fout.write(content)
            except Exception as e:
                logger.info(f"Write file failed. Exception: {e}")
                raise HTTPException(status_code=500, detail=f"Write file {save_path} failed. Exception: {e}")
```